pkg/ingester/instance.go

```diff
@@ -321,6 +323,11 @@ func (i *instance) Query(ctx context.Context, req logql.SelectLogParams) (iter.E
 		return nil, err
 	}
 
+	pipeline, chunkFilter, err := setupDelete(req, pipeline)
+	if i.chunkFilter != nil {
```

Comment:
FYI: chunkFilter is used for LBAC in GEL. So it should definitely be a list of chunkFilters that get applied in order.

Reply:
+1 on this. you should wrap the current chunkFilter if it's not nil.